src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -431,11 +444,29 @@ private HashAlgorithm GetHashAlgorithm(string algorithm)
             {
                 switch (algorithm)
                 {
-                    case "SHA256":
-                    case JwtAlgorithms.RSA_SHA256:
-                    case JwtAlgorithms.ECDSA_SHA256:
-                    case JwtAlgorithms.HMAC_SHA256:
+                    case SecurityAlgorithms.RsaSha1Signature:
+                        return SHA1.Create();
+
+                    case SecurityAlgorithms.SHA256:
+                    case SecurityAlgorithms.ECDSA_SHA256:
+                    case SecurityAlgorithms.HMAC_SHA256:
+                    case SecurityAlgorithms.RSA_SHA256:
+                    case SecurityAlgorithms.RsaSha256Signature:
                         return SHA256.Create();
+
+                    case SecurityAlgorithms.SHA384:
+                    case SecurityAlgorithms.ECDSA_SHA384:
+                    case SecurityAlgorithms.HMAC_SHA384:
+                    case SecurityAlgorithms.RSA_SHA384:
+                    case SecurityAlgorithms.RsaSha384Signature:
+                        return SHA384.Create();
+
+                    case SecurityAlgorithms.SHA512:
+                    case SecurityAlgorithms.RsaSha512Signature:
+                    case SecurityAlgorithms.RSA_SHA512:
+                    case SecurityAlgorithms.ECDSA_SHA512:
+                    case SecurityAlgorithms.HMAC_SHA512:
+                        return SHA512.Create();
```

Comment:
We need to support alg values, PS256, PS384, PS512.  We should augment SecurityAlgorithms with three new values and support them here. Since this only checks the 'hash', even though we may not have support for PS256 (right now) if users have extended crypto, then this would fail and they would have to extend here also to make that scenario supported.